### PR TITLE
Refactor CEED coefficient handling

### PR DESCRIPTION
**Pull request by pazner**

### CODE REVIEW

File: fem/libceed/mf_integrator.hpp [hunk 1]
```diff
@@ -202,54 +149,22 @@ class CeedMFIntegrator : public MFEMCeedOperator
...
-         break;
+         InitCeedBasisAndRestriction(*gridCoeff->coeff->FESpace(), irm, ceed,
+                                       &gridCoeff->basis, &gridCoeff->restr);
+         CeedOperatorSetField(oper, "coeff", gridCoeff->restr,
+                              gridCoeff->basis, gridCoeff->coeffVector);
+      }
+      else if (CeedQuadCoeff *quadCoeff = dynamic_cast<CeedQuadCoeff*>(coeff))
+      {
+         const int ncomp = quadCoeff->ncomp;
+         CeedInt strides[3] = {ncomp, nqpts, ncomp*nqpts};
```
Comment (YohannDudouit) [inline fem/libceed/mf_integrator.hpp:162]:
```suggestion
         CeedInt strides[3] = {ncomp, 1, ncomp*nqpts};
```

**Review by YohannDudouit — APPROVED** [review]
I really like what you did, thank you for doing this.

@barker29 Any objection to merging this in #1924 ? (I think this solves your concern about the `Coefficient` very elegantly)

**Comment by barker29** [comment]
No objection from me, this looks good.

**Comment by tzanio** [comment]
Can we delete the `ceed-refactor-coeff` branch?

**Pull request merged**